src/theory/bv/rewrites

```diff
@@ -692,10 +705,6 @@
   ((x ?BitVec) (n Int))
   (bvslt x (bv 0 n))
   (= (extract (- n 1) (- n 1) x) (bv 1 1)))
-(define-rule bv-zero-ult
```

Comment:
This rule is deleted but there is a regression test added to it. Is it subsumed by another rule?

Reply:
Yes. If you run the mentioned regression test, you get
```lisp
(assume a0 (not (= (bvult #b00000 x) (not (= x #b00000)))))
(step t1 (cl (not (= (not (= (bvult #b00000 x) (not (= x #b00000)))) false)) (not (not (= (bvult #b00000 x) (not (= x #b00000))))) false) :rule equiv_pos2)
(step t2 (cl (= (bvult (bv 0 5) x) (not (= x (bv 0 5))))) :rule all_simplify :args (bv-ult-zero-1 x 5))
(step t3 (cl (= (distinct (bv 0 5) x) (not (= (bv 0 5) x)))) :rule all_simplify :args (distinct-binary-elim (bv 0 5) x))
(step t4 (cl (= (= (bv 0 5) x) (= x (bv 0 5)))) :rule all_simplify :args (eq-symm (bv 0 5) x))
(step t5 (cl (= (not (= (bv 0 5) x)) (not (= x (bv 0 5))))) :rule cong :premises (t4))
(step t6 (cl (= (distinct (bv 0 5) x) (not (= x (bv 0 5))))) :rule trans :premises (t3 t5))
(step t7 (cl (= (not (= x (bv 0 5))) (distinct (bv 0 5) x))) :rule symm :premises (t6))
(step t8 (cl (= (bvult (bv 0 5) x) (distinct (bv 0 5) x))) :rule trans :premises (t2 t7))
(step t9 (cl (= (bvult #b00000 x) (distinct #b00000 x))) :rule hole :premises (t8) :args (ENCODE_PRED_TRANSFORM (= (bvult #b00000 x) (distinct #b00000 x))))
(step t10 (cl (= (distinct (bv 0 5) x) (not (= (bv 0 5) x)))) :rule all_simplify :args (distinct-binary-elim (bv 0 5) x))
(step t11 (cl (= (distinct #b00000 x) (not (= #b00000 x)))) :rule hole :premises (t10) :args (ENCODE_PRED_TRANSFORM (= (distinct #b00000 x) (not (= #b00000 x)))))
(step t12 (cl (= (= (bv 0 5) x) (= x (bv 0 5)))) :rule all_simplify :args (eq-symm (bv 0 5) x))
(step t13 (cl (= (= #b00000 x) (= x #b00000))) :rule hole :premises (t12) :args (ENCODE_PRED_TRANSFORM (= (= #b00000 x) (= x #b00000))))
(step t14 (cl (= (not (= #b00000 x)) (not (= x #b00000)))) :rule cong :premises (t13))
(step t15 (cl (= (distinct #b00000 x) (not (= x #b00000)))) :rule trans :premises (t11 t14))
(step t16 (cl (= (bvult #b00000 x) (not (= x #b00000)))) :rule trans :premises (t9 t15))
(step t17 (cl (= (not (= x #b00000)) (not (= x #b00000)))) :rule refl)
(step t18 (cl (= (= (bvult #b00000 x) (not (= x #b00000))) (= (not (= x #b00000)) (not (= x #b00000))))) :rule cong :premises (t16 t17))
(step t19 (cl (= (= (not (= x (bv 0 5))) (not (= x (bv 0 5)))) true)) :rule all_simplify :args (eq-refl (not (= x (bv 0 5)))))
(step t20 (cl (= (= (not (= x #b00000)) (not (= x #b00000))) true)) :rule hole :premises (t19) :args (ENCODE_PRED_TRANSFORM (= (= (not (= x #b00000)) (not (= x #b00000))) true)))
(step t21 (cl (= (= (bvult #b00000 x) (not (= x #b00000))) true)) :rule trans :premises (t18 t20))
(step t22 (cl (= (not (= (bvult #b00000 x) (not (= x #b00000)))) (not true))) :rule cong :premises (t21))
(step t23 (cl (= (not true) false)) :rule all_simplify :args (evaluate))
(step t24 (cl (= (not (= (bvult #b00000 x) (not (= x #b00000)))) false)) :rule trans :premises (t22 t23))
(step t25 (cl false) :rule resolution :premises (t1 t24 a0))
(step t26 (cl (not false)) :rule false)
(step t27 (cl) :rule resolution :premises (t25 t26))
```

Reply:
Hmm... I am leaning towards renaming the regression test then. If there is no rule with this name it might be confusing? Should stay in in any case!

Reply:
@Lachnitt There's a bv rewrite rule called `ZeroUlt`, and this is where the test got its name.

Reply:
Ah okay thanks that makes sense!